data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -418,4 +425,168 @@ private static String readErrorMessage(InputStream errorStream) throws IOExcepti
+                topicConfig.getAutoCommitInterval().toSecondsPart());
+        properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+                topicConfig.getAutoOffsetReset());
+        properties.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG,
+                topicConfig.getConsumerMaxPollRecords());
+        properties.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, topicConfig.getSessionTimeOut());
+        properties.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, topicConfig.getHeartBeatInterval().toSecondsPart());
+        properties.put(ConsumerConfig.FETCH_MAX_BYTES_CONFIG, topicConfig.getFetchMaxBytes().intValue());
+        properties.put(ConsumerConfig.FETCH_MAX_WAIT_MS_CONFIG, topicConfig.getFetchMaxWait());
+    }
+
+    private void setPropertiesForSchemaRegistryConnectivity(Properties properties) {
+        AuthConfig authConfig = sourceConfig.getAuthConfig();
+        String schemaRegistryApiKey = sourceConfig.getSchemaConfig().getSchemaRegistryApiKey();
+        String schemaRegistryApiSecret = sourceConfig.getSchemaConfig().getSchemaRegistryApiSecret();
+        //with plaintext authentication for schema registry
+        if ("USER_INFO".equalsIgnoreCase(sourceConfig.getSchemaConfig().getBasicAuthCredentialsSource())
+                && authConfig.getSaslAuthConfig().getPlainTextAuthConfig() != null) {
+            String schemaBasicAuthUserInfo = schemaRegistryApiKey.concat(":").concat(schemaRegistryApiSecret);
+            properties.put("schema.registry.basic.auth.user.info", schemaBasicAuthUserInfo);
+            properties.put("basic.auth.credentials.source", "USER_INFO");
+        }
+
+        if (authConfig != null && authConfig.getSaslAuthConfig() != null) {
+            PlainTextAuthConfig plainTextAuthConfig = authConfig.getSaslAuthConfig().getPlainTextAuthConfig();
+            OAuthConfig oAuthConfig = authConfig.getSaslAuthConfig().getOAuthConfig();
+            if (plainTextAuthConfig != null) {
+                String sasl_mechanism = plainTextAuthConfig.getSaslMechanism();
+                String protocol = plainTextAuthConfig.getSecurityProtocol();
+                properties.put("sasl.mechanism", sasl_mechanism);
+                properties.put("security.protocol", protocol);
+            } else if (oAuthConfig != null) {
+                properties.put("sasl.mechanism", oAuthConfig.getOauthSaslMechanism());
+                properties.put("security.protocol", oAuthConfig.getOauthSecurityProtocol());
+            }
+        }
+    }
+
+    private void isTopicExists(String topicName, String bootStrapServer, Properties properties) {
+        List<String> bootStrapServers = new ArrayList<>();
+        String servers[];
+        if (bootStrapServer.contains(",")) {
+            servers = bootStrapServer.split(",");
+            bootStrapServers.addAll(Arrays.asList(servers));
+        } else {
+            bootStrapServers.add(bootStrapServer);
+        }
```

Comment:
I don't this is needed. You just pass the `bootStrapServers` as is with comma separated values in it. That should work.

Reply:
This piece of code is required for the current implementation because we use InetAddress.getByName() for checking the server connectivity and  InetAddress.getByName() requires single server IP details at a time.